Source/WebCore/Modules/speech/SpeechSynthesisUtterance.h

```diff
@@ -92,10 +94,33 @@ class WEBCORE_EXPORT SpeechSynthesisUtterance final : public PlatformSpeechSynth
     void refEventTarget() final { ref(); }
     void derefEventTarget() final { deref(); }
 
+    friend class SpeechSynthesisUtteranceActivity;
     RefPtr<PlatformSpeechSynthesisUtterance> m_platformUtterance;
     RefPtr<SpeechSynthesisVoice> m_voice;
     UtteranceCompletionHandler m_completionHandler;
-    bool m_isActiveForEventDispatch { false };
+    unsigned m_activityCountForEventDispatch { 0 };
+};
+
+class SpeechSynthesisUtteranceActivity {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    SpeechSynthesisUtteranceActivity(RefPtr<SpeechSynthesisUtterance>&& utterance)
+        : m_utterance(utterance)
+    {
+        if (m_utterance)
+            m_utterance->incrementActivityCountForEventDispatch();
+    }
+
+    ~SpeechSynthesisUtteranceActivity()
+    {
+        if (m_utterance)
+            m_utterance->decrementActivityCountForEventDispatch();
+    }
+
+    SpeechSynthesisUtterance* utterance() { return m_utterance.get(); }
+
+private:
+    RefPtr<SpeechSynthesisUtterance> m_utterance;
```

Comment:
Can this be Ref instead? It logically doesn't make sense for this object to exist with a null m_utterance, right?